Python/import.c

```diff
@@ -1297,13 +1297,16 @@ find_frozen(PyObject *nameobj, struct frozen_info *info)
         info->nameobj = nameobj;  // borrowed
         info->data = (const char *)p->code;
         info->get_code = p->get_code;
-        info->size = p->size < 0 ? -(p->size) : p->size;
-        info->is_package = p->size < 0 ? true : false;
+        info->size = p->size;
+        info->is_package = p->is_package;
```

Comment:
To play it safe for backward compatibility, consider setting `is_package` to true if `size` is negative.

```suggestion
        if (p->size < 0) {
            // backward compatibility
            info->size = -(p->size);
            info->is_package = true;
        } else {
            info->size = p->size;
            info->is_package = p->is_package;
        }
```

or even

```suggestion
        info->size = p->size;
        info->is_package = p->is_package;
        if (p->size < 0) {
            // backward compatibility
            info->size = -(p->size);
            info->is_package = true;            
        }
```

Reply:
Fixed